cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/BatchSQLEngineAdapter.java

```diff
@@ -293,8 +298,9 @@ public boolean canJoin(String datasetName,
         }
 
         joinInternal(future, joinRequest);
-      } catch (Exception e) {
-        future.completeExceptionally(e);
+        LOG.debug("Completed join for dataset '{}'", datasetName);
+      } catch (Exception t) {
```

Comment:
You need throwable here as well, don't you?

Reply:
My bad, you are correct